backend/btrixcloud/crawls.py

```diff
@@ -829,17 +801,48 @@ async def add_or_remove_exclusion(self, crawl_id, regex, org, user, add):
 
         return resp
 
-    async def remove_crawl_from_collections(self, org: Organization, crawl_id: str):
-        """Remove crawl with given crawl_id from all collections it belongs to"""
-        collections = [
-            coll["name"]
-            async for coll in self.collections.find({"crawlIds": {"$in": [crawl_id]}})
-        ]
-        for collection_name in collections:
-            await self.collections.find_one_and_update(
-                {"name": collection_name, "oid": org.id},
-                {"$pull": {"crawlIds": crawl_id}},
-            )
+    async def add_to_collection(
+        self, crawl_id: uuid.UUID, collection_id: uuid.UUID, org: Organization
+    ):
+        """Add crawl to collection."""
+        crawl_raw = await self.get_crawl_raw(crawl_id, org)
+        if crawl_id in crawl_raw.get("collections"):
+            raise HTTPException(status_code=400, detail="crawl_already_in_collection")
+
+        result = await self.crawls.find_one_and_update(
+            {"_id": crawl_id},
+            {"$push": {"collections": collection_id}},
+            return_document=pymongo.ReturnDocument.AFTER,
+        )
+        if not result:
+            raise HTTPException(status_code=404, detail="crawl_not_found")
+
+    async def remove_from_collection(
+        self, crawl_id: uuid.UUID, collection_id: uuid.UUID
+    ):
+        """Remove crawl from collection."""
+        result = await self.crawls.find_one_and_update(
+            {"_id": crawl_id},
+            {"$pull": {"collections": collection_id}},
+            return_document=pymongo.ReturnDocument.AFTER,
+        )
+        if not result:
+            raise HTTPException(status_code=404, detail="crawl_not_found")
+
+    async def remove_collection_from_all_crawls(self, collection_id: uuid.UUID):
+        """Remove collection id from all crawls it's currently in."""
+        result = await self.crawls.update_many(
+            {"collections": collection_id},
+            {"$pull": {"collections": collection_id}},
+        )
+        if result.modified_count < 1:
+            raise HTTPException(status_code=404, detail="crawls_not_found")
+
+    async def get_crawls_in_collection(self, collection_id: uuid.UUID):
```

Comment:
Does it make sense to have this separate from the more generic `list_crawls`, which has a bunch of other options too?
I guess this is for no pagination + sort order irrelevant query, such as replay?

Reply:
Yes, exactly - it's a faster and simpler lookup to just get a list of crawl ids related to a collection, without needing to worry about pagination. It's also useful for annotating collections for `CollOut` (page count and tags coming soon will both use it)